b2sdk/raw_api.py

```diff
@@ -914,16 +948,23 @@ def copy_file(
             kwargs['sourceServerSideEncryption'
                   ] = source_server_side_encryption.serialize_to_json_for_request()
 
-        # FIXME: implement `legal_hold` and `file_retention`
+        if legal_hold is not None:
+            kwargs['legalHold'] = legal_hold.to_server()
 
-        return self._post_json(
-            api_url,
-            'b2_copy_file',
-            account_auth_token,
-            sourceFileId=source_file_id,
-            fileName=new_file_name,
-            **kwargs
-        )
+        if file_retention is not None:
+            kwargs['fileRetention'] = file_retention.serialize_to_json_for_request()
+
+        try:
+            return self._post_json(
+                api_url,
+                'b2_copy_file',
+                account_auth_token,
+                sourceFileId=source_file_id,
+                fileName=new_file_name,
+                **kwargs
+            )
+        except AccessDenied:
+            raise SSECKeyError()
```

Comment:
If you would have correct SSEC key - but no write legal hold/file retention capability wouldn't it raise wrong error?